File: UploadDownload/src/main/resources/static/app/services/upload.service.js
(function () {
    'use strict';
    angular
            .module('appUpload')
            .service('UploadService', function ($http) {
                var self = this;

                self.getUploadr = function () {
                    return $http({
                        method: 'GET',
                        url: 'api/upload',
                        headers: {
                            'Content-Type': 'application/json'

                        }
                    });
                };

                self.save = function (upload) {
                    return $http({
                        method: 'POST',
                        url: 'api/upload',
                        headers: {
                            'Content-Type': 'application/json'

                        },
                        data: upload
                    });
                };

                self.update = function (upload) {
                    return $http({
                        method: 'PUT',
                        url: 'api/upload/' + upload.idUpload,
                        headers: {
                            'Content-Type': 'application/json'
                        },
                        data: upload
                    });
                };

                self.delete = function (upload) {
                    return $http({
                        method: 'DELETE',
                        url: 'api/upload/' + upload.idUpload,
                        headers: {
                            'Content-Type': 'application/json'
                        }
                    });
                };



            });
})();

File: UploadDownload/src/main/java/com/lowongan/telkom/service/UploadService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.lowongan.telkom.service;

import com.lowongan.telkom.domain.Upload;
import java.util.Optional;

/**
 *
 * @author Irman Juliansyah <irmanjuliansyah@gmail.com>
 */
public interface UploadService {
    
    Upload save (Upload upload);
    
    Upload update (Upload upload);
    
    public void delete (String idUpload);
    
    Optional<Upload> getUpload(String idUpload);
    
    Iterable<Upload> getUploads();
}
